src/Arcus.Messaging.Abstractions/MessageCorrelationInfo.cs

```diff
@@ -4,31 +4,41 @@
+    /// Represents the information concerning correlation of telemetry &amp; processes with main focus on messaging scenarios.
     /// </summary>
     public class MessageCorrelationInfo : CorrelationInfo
     {
         /// <summary>
-        ///     Unique identifier that indicates an attempt to process a given message
+        /// Initializes a new instance of the <see cref="MessageCorrelationInfo"/> class.
         /// </summary>
-        /// <remarks>
-        ///     If the same message is processed n-times,
-        ///     it will have the same OperationId but n different cycle ids
-        /// </remarks>
-        public string CycleId { get; }
+        /// <param name="operationId">The unique identifier that spans one operation end-to-end.</param>
+        /// <param name="transactionId">The unique identifier that spans one or more operations and are considered a transaction/session.</param>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="operationId"/> is blank.</exception>
+        public MessageCorrelationInfo(string operationId, string transactionId)
+            : base(operationId, transactionId, operationParentId: null)
+        {
+            CycleId = Guid.NewGuid().ToString("D");
+        }
 
         /// <summary>
-        ///     Constructor
+        /// Initializes a new instance of the <see cref="MessageCorrelationInfo" /> class.
         /// </summary>
-        /// <param name="transactionId">
-        ///     Unique identifier that spans one or more operations and are considered a
-        ///     transaction/session
-        /// </param>
-        /// <param name="operationId">Unique identifier that spans one operation end-to-end</param>
-        public MessageCorrelationInfo(string operationId, string transactionId)
-            : base(operationId, transactionId)
+        /// <param name="operationId">The unique identifier that spans one operation end-to-end.</param>
+        /// <param name="transactionId">The unique identifier that spans one or more operations and are considered a transaction/session.</param>
+        /// <param name="operationParentId">The unique identifier of the original service that initiated this request.</param>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="operationId"/> is blank.</exception>
+        public MessageCorrelationInfo(string operationId, string transactionId, string operationParentId)
+            : base(operationId, transactionId, operationParentId)
         {
             CycleId = Guid.NewGuid().ToString("D");
         }
+
+        /// <summary>
+        /// Gets the unique identifier that indicates an attempt to process a given message.
+        /// </summary>
+        /// <remarks>
+        ///     If the same message is processed n-times, it will have the same OperationId but n different cycle ids
+        /// </remarks>
+        public string CycleId { get; }
```

Comment:
It would be a breaking change, though. It's still more an ID than a counter as it's more linked to the message pump/processing than the amount of times it's processed. So, it's more about the 'cycle', then, maybe.